### DISCUSSION TITLE
wait before execute?

### DISCUSSION THREAD

**Discussion by mawnir** [pmndrs/zustand#2104] [Q&A] [answered]
hi,
i have an [] called 'users' in my store..  to use it:

```
const  { users } = useLoginStore((state) => state);
if (users.length == 0) {
        redirect('/authentication/sign-in'); => this get executed
} 
```
i think users.length returns 0 at first, because I'm sure users.length is 1.
i hope i explained myself correctly.
how can i fix it?

**Comment by dai-shi** [comment]
Not sure if I get the use case nor the setup, but how about using `useEffect`?

**Reply by mawnir** [reply]
hi, this is a small version of my store:

```
const loginStore = (set) => ({
    users: [],
    login: (user) => set((state) => ({ users: [...state.users, user] })),
    logout: (email) =>
        set((state) => ({
            users: state.users.filter((user) => user.email !== email),
        })),
});
```
my use case(layout.js):  
```
export default function DashboardLayout({ children }) {

    const { users } = useLoginStore((state) => state)

    if (!users || users.length == 0) {
          redirect('/authentication/sign-in');
      }

    return (
    // My Dashboard
    );
}
```
Thank you.. `useEffect` has fixed the issue. but i prefer not to use it, because the page loads the content, then redirect.

**Reply by dai-shi** [reply, marked as answer]
In that case, you want to have a pending state. (I don't know if `redirect` should be in useEffect or in render, anyways.)

Your initial state is `users=[]`, so you can't distinguish it from the logged-out state.

**Reply by mawnir** [reply]
a pending state? like this: 
```
  return (
   {condition?
    // My Dashboard : <Not logged>
   }
    );
```

**Reply by dai-shi** [reply]
Yeah and like:
```js
const useLoginStore = create((set) => ({
  usersInitialized: false,
  users: [],
  // ...
}));
```

**Reply by mawnir** [reply]
Thank you
